sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileFormatDataWriter.scala

```diff
@@ -186,7 +186,7 @@ class GpuSingleDirectoryDataWriter(
             tabs.foreach(b => {
               if (needNewWriter) {
                 fileCounter += 1
-                assert(fileCounter < MAX_FILE_COUNTER,
+                assert(fileCounter <= MAX_FILE_COUNTER,
```

Comment:
to keep things conventional, it might be better to make the counter 0-based instead of fixing the assert

Reply:
The counter base is not changed and the value of `file_counter` always starts from `0` which indicates that we don't write any file.
I guess this was bug that the `MAX_FILE_COUNTER` should be inclusive, which you can find from the error message `s"File counter ${currentWriterStatus.fileCounter} " +
                s"is beyond max value $MAX_FILE_COUNTER")` that we should only give this error when `fileCounter > MAX_FILE_COUNTER`

Reply:
This PR aims to enable `MaxRecordsPerFile` config.
The MAX_FILE_COUNTER is another issue, let's keep it as the previous `<`?
And we should check the Spark behavior before we change it.
There is `fileCounter < MAX_FILE_COUNTER` in `GpuSingleDirectoryDataWriter`, let's find all of them and handle them together.